### DISCUSSION TITLE
Issue while serving the apple wallet pass

### DISCUSSION THREAD

**Discussion by prashant03Dev** [alexandercerutti/passkit-generator#200] [Q&A] [answered]
@alexandercerutti Suggest me the way to share the pass to the device using webserviceUrl,and how will we get the webserviceUrl,

As you refered me the docs for apple wallet documentation,I'm providing the link below down.

https://developer.apple.com/documentation/walletpasses/adding_a_web_service_to_update_passes

**Comment by alexandercerutti** [comment, marked as answer]
> Suggest me the way

Am I now obliged to do it?

What do you mean, by the way? Describe your use case please.

**Reply by prashant03Dev** [reply]
@alexandercerutti 

As i have pipe the buffer and send it to frontend,but when do we know that the user is requesting for the pass update,and the backend will re-generate it.

**Reply by alexandercerutti** [reply]
As you could read in the documentation and in my other projects, you can use the "Send an Updated Pass" endpoint: https://developer.apple.com/documentation/walletpasses/send_an_updated_pass

I've implemented a sample server here with my plugin: https://github.com/alexandercerutti/fastify-passkit-webservice/blob/master/specs/server.mjs
